doc/getting_started.rst

```diff
@@ -0,0 +1,230 @@
+
+- The samples matrix (or design matrix) :term:`X`. The size of ``X``
+  is typically ``(n_samples, n_features)``, which means that samples are
+  represented as rows and features are represented as columns.
+- The target values :term:`y` which are real numbers for regression tasks, or
+  integers for classification (they can also be strings). For unsupervized
+  learning tasks, ``y`` does not need to be specified. ``y`` is usually 1d
+  array where the ``i`` th entry corresponds to the target of the ``i`` th
+  sample (row) of ``X``.
+
+Both ``X`` and ``y`` are usually expected to be numpy arrays or equivalent
+:term:`array-like`, data types, though some estimators work with other
+formats such as sparse matrices.
+
+Once the estimator is fitted, it can be used for predicting target values of
+new data. You don't need to re-train the estimator::
+
+  >>> clf.predict(X)  # predict classes of the training data
+  array([0, 1])
+  >>> clf.predict([[4, 5, 6], [14, 15, 16]])  # predict classes of new data
+  array([0, 1])
+
+Transformers and pre-processors
+-------------------------------
+
+Machine learning worflows are often composed of different parts. A typical
+pipeline consists of a pre-processing step that transforms or impute the
+data, and a final predictor that predicts target values.
+
+In ``scikit-learn``, pre-processors and transformers follow the same API as
+the estimator objects (they actually all inherit from the same
+``BaseEstimator`` class). The transformer objects don't have a
+:term:`predict` method but rather a :term:`transform` method that outputs a
+newly transformed sample matrix ``X``::
+
+  >>> from sklearn.preprocessing import StandardScaler
+  >>> X = [[0, 15],
+  ...      [1, -10]]
+  >>> StandardScaler().fit(X).transform(X)
+  array([[-1.,  1.],
+         [ 1., -1.]])
+
+Sometimes, you want to apply different transformations to different features:
+the :ref:`ColumnTransformer<column_transformer>` is designed for these
+use-cases.
+
+Pipelines: chaining pre-preocessors and estimators
+--------------------------------------------------
+
+Transformers and estimators (predictors) can be combined together into a
+single unifying object: a :class:`~sklearn.pipeline.Pipeline`. The pipeline
+offers the same API as a regular estimator: it can be fitted and used for
+prediction with ``fit`` and ``predict``. As we will see later, using a
+pipeline will also prevent you from disclosing some testing data in your
+training data.
+
+In the following example, we :ref:`load the Iris dataset <datasets>`, split it
+into train and test sets, and compute the accuracy score of a pipeline on
+the test data::
+
+  >>> from sklearn.preprocessing import StandardScaler
+  >>> from sklearn.linear_model import LogisticRegression
+  >>> from sklearn.pipeline import make_pipeline
+  >>> from sklearn.datasets import load_iris
+  >>> from sklearn.model_selection import train_test_split
+  >>> from sklearn.metrics import accuracy_score
+  ...
```

Comment:
it's interesting that the doctest tests pass here. I remember they really needed a space after `...`